src/registry/mod.rs

```diff
@@ -108,4 +109,41 @@ impl RegistryUri {
 
         Ok(())
     }
+
+    /// Returns the type of registry this URI refers to
+    pub fn kind(&self) -> RegistryType {
+        RegistryType::Artifactory
+    }
+}
+
+#[derive(Default)]
+pub struct RegistryProvider {
+    credentials: Credentials,
+    registries: HashMap<RegistryUri, Arc<dyn Registry + Send + Sync>>,
```

Comment:
this is overly complex, as we just have artifactory currently?

and if we move away from artifactory, we will have _exactly_ one network registry impl (the buffrs registry)

Reply:
```suggestion
    registries: HashMap<RegistryUri, Arc<R>>,
```

Reply:
I know, but my suggested change is inline with what you wrote, only narrowing the solution to we "just have one registry backend" axiom